plugins/modules/clustering/consul/consul.py

```diff
@@ -343,63 +343,61 @@ def remove_service(module, service_id):
     consul_api = get_consul_api(module)
     service = get_service_by_id_or_name(consul_api, service_id)
     if service:
-        consul_api.agent.service.deregister(service_id, token=module.params.get('token'))
+        consul_api.agent.service.deregister(service_id, token=module.params['token'])
         module.exit_json(changed=True, id=service_id)
 
     module.exit_json(changed=False, id=service_id)
 
 
 def get_consul_api(module):
-    consulClient = consul.Consul(host=module.params.get('host'),
-                                 port=module.params.get('port'),
-                                 scheme=module.params.get('scheme'),
-                                 verify=module.params.get('validate_certs'),
-                                 token=module.params.get('token'))
+    consulClient = consul.Consul(host=module.params['host'],
+                                 port=module.params['port'],
+                                 scheme=module.params['scheme'],
+                                 verify=module.params['validate_certs'],
+                                 token=module.params['token'])
     consulClient.agent.service = PatchedConsulAgentService(consulClient)
     return consulClient
 
 
 def get_service_by_id_or_name(consul_api, service_id_or_name):
     ''' iterate the registered services and find one with the given id '''
-    for name, service in consul_api.agent.services().items():
-        if service['ID'] == service_id_or_name or service['Service'] == service_id_or_name:
+    for dummy, service in consul_api.agent.services().items():
+        if service_id_or_name in [service['ID'], service['Service']]:
             return ConsulService(loaded=service)
 
 
 def parse_check(module):
-    if len([p for p in (module.params.get('script'), module.params.get('ttl'), module.params.get('tcp'), module.params.get('http')) if p]) > 1:
+    _checks = [module.params[p] for p in ('script', 'ttl', 'tcp', 'http')]
```

Comment:
Why not already collect the ones which evaluate to `True`? In both the next `if` you need exactly these, and not the others.

Reply:
yup, makes more sense